### PR TITLE
add: parameter injection pattern

### PR DESCRIPTION
**Pull request by fkromer**
refactored from constructor injection pattern, cleanup

### CODE REVIEW

**Comment by fkromer** [comment]
ready for review

File: dft/parameter_injection.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+production code which is untestable:
+
+class TimeDisplay(object):
+
+    def __init__(self):
+        self.time_provider = datetime.datetime
+
+    def get_current_time_as_html_fragment(self):
+        current_time = self.time_provider.now()
+        current_time_as_html_fragment = "<span class=\"tinyBoldText\">" + current_time.hour + ":" + current_time.minute + "</span>"
```
Comment (faif) [inline dft/parameter_injection.py]: Same comment here: Use `.format` which is more idiomatic than `+`

Comment (fkromer) [inline dft/parameter_injection.py]: DONE

**Comment by faif** [comment]
There are conflicts. Can you please fix them?

**Comment by fkromer** [comment]
@faif Thanks for merging, but I miss `parameter_injection.py` [here](https://github.com/faif/python-patterns/tree/master/dft) and `test_parameter_injection.py` [here](https://github.com/faif/python-patterns/tree/master/tests). (I just recognized it when I was searching for it to hyperlink it...)

**Comment by faif** [comment]
Can you create a new pull request including those changes?

Thanks

**Pull request merged**